### PR TITLE
feat(Created-Patch 2):Created API for patch2 and created frontend components for the same

### PR DESCRIPTION
**Pull request by Kanishk6677**
Signed-off-by: Kanishk6677 <kanishkraj6677@gmail.com>

### Functionality:
Patch 2

### Solution:
Get the field name and field value from the user to get task details based on it and update the fields with new values in all other tasks too.

### Risk level:
- [ ] high 
- [x] medium
- [ ] low

### How to test:
1. firebase emulators : start
2. ng serve

### CODE REVIEW

**Review by RohitVarma001 — COMMENTED** [review]
Ship it

**Comment by k0malSharma** [comment]
Resolve Conflict ASAP

File: functions/patch2API.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        Promise.all(Promises).then(() => {
+            result = { data: "OK! Patch2 executed" };
+            console.log("Counters updated");
+            return response.status(200).send(result);
+        }).catch(function(error) {
+            result = { data: error };
+            console.error("Patch error in updating counters", error);
+            return response.status(500).send(result);
+        });
+        });
```
Comment (viv9k) [inline functions/patch2API.js:42]: I think this an extra closing, please fix it.

File: functions/patch2API.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        const orgDomain = request.body.data.OrgDomain;
+        let taskId;
+        const newfield=request.body.data.newField;
+        const newFieldValue=request.body.data.NewFieldValue;
+        const promise1=db.collection("Organizations").doc(orgDomain).collection("Tasks").get().then((task)=>{
+            task.forEach((doc)=>{
+                taskId = doc.data().Id;
+                console.log("Executing Promise1 of Patch2");
+                desiredField={};
+                desiredField[newfield]=newFieldValue;
```
Comment (viv9k) [inline functions/patch2API.js:28]: renaming desiredField as data could sound much better.

**Review by viv9k — CHANGES_REQUESTED** [review]
Please complete the required changes

**Review by viv9k — APPROVED** [review]
Ship it

**Pull request merged**